jlibra-core/src/main/java/dev/jlibra/admissioncontrol/query/TransactionListWithProof.java

```diff
@@ -0,0 +1,28 @@
+package dev.jlibra.admissioncontrol.query;
+
+import org.immutables.value.Value;
+import types.Events;
+
+import java.util.Collection;
+import java.util.List;
+
+import static java.util.stream.Collectors.toList;
+
+@Value.Immutable
+public interface TransactionListWithProof {
+    List<Event> getEvents();
+
+    static TransactionWithProof fromGrpcObject(types.TransactionOuterClass.TransactionListWithProof grpcObject) {
+        List<List<Events.Event>> collect = grpcObject.getEventsForVersions().getEventsForVersionList().stream()
+                .map(Events.EventsList::getEventsList)
+                .collect(toList());
```

Comment:
I find it unnecessary to `collect()` only to immediately `stream()` it again. Both operations can be dropped, simply continue transforming

Reply:
Adjusted accordingly